### PR TITLE
fix softmax with cross entropy out of bound

### PR DESCRIPTION
**Pull request by phlrain**
### PR types
Bug fixes
### PR changes
OPs

### Describe
修复 softmax with cross entropy 显存访问越界的问题，如果label中有一个id 大于待处理的维度， 会发生访问越界的情况

例如logits.shape = [ 3, 5], labels= [ 1, 2, 10], ignore_index = 10, 当处理lables中的10是，是大于5的，会发生写越界

### CODE REVIEW

**Review by Xreki — APPROVED** [review]
LGTM
